include/abb_librws/common/rw/elog.h

```diff
@@ -0,0 +1,123 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include <chrono>
+
+namespace abb ::rws ::rw ::elog
+{
+    // TODO: Replace with class that pulls domains?
+    /**
+     * @brief Elog domain enum
+     */
+    enum ElogDomain
+    {
+        COMMON = 0,
+        OPERATIONAL = 1,
+        SYSTEM = 2,
+        HARDWARE = 3,
+        PROGRAM = 4,
+        MOTION = 5,
+        IO_AND_COMMUNICATION = 7,
+        USER = 8,
+        SAFETY = 9,
+        INTERNAL = 10,
+        PROCESS = 11,
+        CONFIGURATION = 12,
+        RAPID_1 = 15,
+        RAPID_2 = 17
+    };
+
+    std::string elogDomainToString(ElogDomain domain);
+
+
+    /**
+     * @brief Types of elog messages
+     */
+    enum MessageType
+    {
+        INFORMATION = 1,
+        WARNING = 2,
+        ERROR = 3
+    };
+
+    std::string messageTypeToString(MessageType const messagType);
+
+
+    /**
+     * @brief Types of elog message arguments
+     */
+    enum ArgType
+    {
+        STRING,
+        LONG,
+        FLOAT
+    };
+
+    ArgType makeArgType(std::string const &str);
+
+    std::string argTypeToString(ArgType const &argType);
+
+    
+    /**
+     * @brief Elog message argument
+     */
+    struct ElogMessageArg
+    {
+        int argNum;
+        ArgType argType;
+        std::string value;
+
+        float toFloat();
+        long toLong();
+
+        std::string toString();
+    };
+
+
+    /**
+     * @brief Elog message structure
+     */
+    struct ElogMessage
+    {
+        int domain;                       // Domain of message.
+        int sequenceNumber;               // Sequence number of message in domain.
+        MessageType messageType;          // The elog message type.
+        int code;                         // The elog message code.
+        std::string sourceName;           // The elog message source.
+        std::tm timestamp;                // The time stamp when the event log was generated.
+        std::string title;                // The elog message title in the specified langauge. Available only when lang parameter is provided.
+        std::string desc;                 // The elog message description in the specified langauge. Available only when lang parameter is provided.
+        std::string conseqs;              // The elog message consequences in the specified langauge. Available only when lang parameter is provided.
+        std::string causes;               // The elog message causes in the specified langauge. Available only when lang parameter is provided.
+        std::string actions;              // The elog message actions in the specified langauge. Available only when lang parameter is provided.
+        std::vector<ElogMessageArg> argv; // The argument's position e.g. arg1, arg2 etc.. The type of argument can be float, string or long.
+
+        std::string toString();
```

Comment:
As a general rule, one should not declare as a class/struct member what does not need access to class private data. Consider adding a free `to_string()` function or overloading `operator<<()`.